router/errors.go

```diff
@@ -85,7 +86,8 @@ func httpError(code int, fmtString string, args ...interface{}) *HTTPError {
 
 // HandleError will handle an error
 func HandleError(err error, w http.ResponseWriter, r *http.Request) {
-	if err == nil {
+	// if err is nil or nil pointer of type *HTTPError
+	if err == nil || (*[2]uintptr)(unsafe.Pointer(&err))[1] == 0 {
```

Comment:
if err.(*HTTPError) == nil || err == nil will also work and then you don't have to use unsafe

Reply:
This introduces unsafe into the netlify-commons package which is used by most services. therefore introducing this in them as well.  I checked the code and it is valid but there is a better way of doing this with type assertion.  You can also run go vet to check if we are doing something not so smart with unsafe.  Unsafe steps around Go's type safety and in my opinion should only be used way no other way is possible

Reply:
Agree on not using `unsafe` if we can avoid it

Reply:
The issue is not about using `unsafe`. The issue is the function signature and the responsibilities of it.  The function could receive **any** implementation of `error` that ends up with the same problem.

Btw, I recommend you to run a grep on all dependencies looking for the usage of `unsafe`. 
Feel free to open a new PR with the needed changes if you agree.